tests/lib/newlib/newlib_hook/testcase.yaml

```diff
@@ -0,0 +1,10 @@
+tests:
+  libraries.libc.newlib.userspace:
+    platform_exclude: mps2_an521_ns
```

Comment:
Why are we excluding `mps2_an521_ns`?

Reply:
In userspace test, here is a build error in build.log on mps2_an521_ns.  I don't think this is my code problem.

 91%] Building C object bl2/ext/mcuboot/CMakeFiles/signing_layout_s.dir/signing_layout_s.o
[ 91%] Built target signing_layout_s
[ 92%] Building C object bl2/ext/mcuboot/CMakeFiles/signing_layout_ns.dir/signing_layout_ns.o
[ 92%] Built target signing_layout_ns
[ 92%] Generating tfm_s_signed.bin
Traceback (most recent call last):
  File "/home/yingming/zephyrproject/modules/tee/tfm/trusted-firmware-m/bl2/ext/mcuboot/scripts/wrapper/wrapper.py", line 21, in <module>
    import imgtool.main
  File "/home/yingming/zephyrproject/bootloader/mcuboot/scripts/imgtool/main.py", line 23, in <module>
    import imgtool.keys as keys
  File "/home/yingming/zephyrproject/bootloader/mcuboot/scripts/imgtool/keys/__init__.py", line 27, in <module>
    from cryptography.hazmat.primitives.asymmetric.ed25519 import (
ModuleNotFoundError: No module named 'cryptography.hazmat.primitives.asymmetric.ed25519'
bl2/ext/mcuboot/CMakeFiles/tfm_s_signed_bin.dir/build.make:70: recipe for target 'bl2/ext/mcuboot/tfm_s_signed.bin' failed
make[5]: *** [bl2/ext/mcuboot/tfm_s_signed.bin] Error 1
CMakeFiles/Makefile2:2096: recipe for target 'bl2/ext/mcuboot/CMakeFiles/tfm_s_signed_bin.dir/all' failed
make[4]: *** [bl2/ext/mcuboot/CMakeFiles/tfm_s_signed_bin.dir/all] Error 2
Makefile:135: recipe for target 'all' failed
make[3]: *** [all] Error 2
modules/trusted-firmware-m/CMakeFiles/tfm.dir/build.make:88: recipe for target 'modules/trusted-firmware-m/tfm-prefix/src/tfm-stamp/tfm-build' failed
make[2]: *** [modules/trusted-firmware-m/tfm-prefix/src/tfm-stamp/tfm-build] Error 2
CMakeFiles/Makefile2:3929: recipe for target 'modules/trusted-firmware-m/CMakeFiles/tfm.dir/all' failed
make[1]: *** [modules/trusted-firmware-m/CMakeFiles/tfm.dir/all] Error 2
Makefile:90: recipe for target 'all' failed
make: *** [all] Error 2

Reply:
After update the environment, this build error disappear.
Just run:
pip3 install --upgrade pip
pip3 install --user -r ~/zephyrproject/zephyr/scripts/requirements.txt